src/wrapper/clap-juce-wrapper.cpp

```diff
@@ -169,20 +169,31 @@ class ClapJuceWrapper : public clap::helpers::Plugin<clap::helpers::Misbehaviour
 #endif
     }
 
+    bool haveCompletedDeferredInit{false};
     bool init() noexcept override
     {
-#if JUCE_LINUX
-        if (_host.canUseTimerSupport())
-        {
-            _host.timerSupportRegister(1000 / 50, &idleTimer);
-        }
-#endif
         defineAudioPorts();
 
+        haveCompletedDeferredInit = false;
+        _host.requestCallback();
         return true;
     }
 
-  public:
+    void onMainThread() noexcept override
+    {
+        if (!haveCompletedDeferredInit)
+        {
+            haveCompletedDeferredInit = true;
+
+#if JUCE_LINUX
+            if (_host.canUseTimerSupport())
+            {
+                _host.timerSupportRegister(1000 / 50, &idleTimer);
```

Comment:
Is there a reason for the number `1000 / 50`?

Reply:
50 times a second seemed ok. I guess I could change it to 60 to be 60hz and will do so